crates/driver/src/boundary/liquidity/zeroex.rs

```diff
@@ -1,11 +1,61 @@
 use {
-    crate::{infra, infra::Ethereum},
+    crate::{
+        domain::{
+            eth,
+            liquidity::{self, zeroex},
+        },
+        infra::{self, Ethereum},
+    },
     ethrpc::current_block::CurrentBlockStream,
-    shared::{http_client::HttpClientFactory, zeroex_api::DefaultZeroExApi},
-    solver::{liquidity::zeroex::ZeroExLiquidity, liquidity_collector::LiquidityCollecting},
+    shared::{
+        http_client::HttpClientFactory,
+        http_solver::model::InternalizationStrategy,
+        price_estimation::gas::GAS_PER_ZEROEX_ORDER,
+        zeroex_api::DefaultZeroExApi,
+    },
+    solver::{
+        liquidity::{zeroex::ZeroExLiquidity, LimitOrder, LimitOrderExecution},
+        liquidity_collector::LiquidityCollecting,
+        settlement::SettlementEncoder,
+    },
```

Comment:
How much work would it be to not depend on this code but instead port what we need into the driver?
While not strictly required to achieve what we need here, it's covered by #2215 so we might as well start it here if it's not too much work.

Reply:
`LiquidityCollecting` is currently only possible to move with since this trait is used for all other liquidity providers. `LimitOrder` is part of it as well too. 

`ZeroExLiquidity` probably can be moved, but it contains a lot of code and dependencies to the `solver` crate. It would be better to try in a separate PR. `SettlementEncoder` and `LimitOrderExecution` are removed.